src/classifier/Pipe.cpp

```diff
@@ -45,6 +49,7 @@ void Pipe::Initialize() {
   CreateWriter();
   CreateDecoder();
   parameters_ = new Parameters;
+  running_multithreaded_ = false;
```

Comment:
Do you need this as a class member variable? It's in Options, right? 

Also, if you need it, rename it "run_multithreads" or "use_multithreads" instead of "running".